.github/workflows/integration.yml

```
@@ -0,0 +1,102 @@
+# Run secret-dependent integration tests only after /ok-to-test approval
+on:
+  pull_request:
+  repository_dispatch:
+    types: [ok-to-test-command]
+
+name: Integration tests
+
+jobs:
+  # Branch-based pull request
+  integration-trusted:
+    runs-on: ubuntu-latest
+    if: github.event_name == 'pull_request' && github.event.pull_request.head.repo.full_name == github.repository
+    strategy:
+      matrix:
+        target:
+          - check-docs
+          - check-mod
+          - lint-ci
+          - test
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-go@v2
+        with:
+          go-version: '1.15.2'
+      - name: Install dependencies
+        run: make setup
+
+      - name: make ${{ matrix.target }}
+        env:
+          REVIEWDOG_GITHUB_API_TOKEN: ${{ secrets.REVIEWDOG_GITHUB_API_TOKEN }}
+          SNOWFLAKE_USER: ${{ secrets.SNOWFLAKE_USER }}
+          SNOWFLAKE_PASSWORD: ${{ secrets.SNOWFLAKE_PASSWORD }}
+          SNOWFLAKE_ACCOUNT: ${{ secrets.SNOWFLAKE_ACCOUNT }}
+          SNOWFLAKE_ROLE: ${{ secrets.SNOWFLAKE_ROLE }}
+        run: make ${{ matrix.target }}
+
+  # Repo owner has commented /ok-to-test on a (fork-based) pull request
+  integration-fork:
+    runs-on: ubuntu-latest
+    if:
+      github.event_name == 'repository_dispatch' &&
+      github.event.client_payload.slash_command.sha == github.event.client_payload.pull_request.head.sha
+    # Integration tests needing secrets
+    strategy:
+      matrix:
+        target:
+          - check-docs
+          - check-mod
+          - lint-ci
+          - test
```

```suggestion
          - test-acceptance-ci
```
is that the target that runs the acceptance tests?

---

https://github.com/chanzuckerberg/terraform-provider-snowflake/pull/320#discussion_r534560084

---

yep, but you still want the `test-acceptance-ci` for the workflow that requires secrets. But only the `test` for the workflow with no secrets. If you look at https://github.com/chanzuckerberg/terraform-provider-snowflake/blob/main/Makefile#L68, it has a `TF_ACC=1` which indicates we can actually go and create real resources (run acceptance tests).

---

I'm actually not super clear why we have both `test-acceptance` and `test-acceptance-ci` I vote we rm `test-acceptance-ci` and just keep `test-acceptance` ?